### PR TITLE
Update SemVerFactory to handle 'v' prefix

### PR DESCRIPTION
**Pull request by scme0**
# Background

I have been looking into a bug where a specific Helm Feed wasn't working correctly and as it turns out, that particular Helm Feed returns HelmCharts with versions with a 'v' prefix eg: `version: "v1.0.0"`. This currently doesn't parse correctly and so when you search for helm charts on this feed, it returns no results (with no errors).

# Solution

I propose that we add support for parsing a semver with a preceding 'v' as I've implemented in this Pull Request. The v is effectively ignored during parsing but it is maintained in the `OriginalString` property.

### CODE REVIEW

File: source/Octopus.Versioning.Tests/Octopus/OctopusVersionTests.cs [hunk 1]
```diff
@@ -777,15 +804,6 @@ public void TryParseTest()
...
             "")]
-        [TestCase("v0.0.1",
-            0,
-            0,
-            1,
-            0,
-            "",
-            "",
-            "",
-            "")]
```
Comment (YuKitsune) [inline source/Octopus.Versioning.Tests/Octopus/OctopusVersionTests.cs:788]: How did these test cases work if we didn't have support for a leading `v` before?

Comment (scme0) [reply source/Octopus.Versioning.Tests/Octopus/OctopusVersionTests.cs:788]: These were in the "Invalid semver" tests, I've moved them to the valid test.

Comment (YuKitsune) [reply source/Octopus.Versioning.Tests/Octopus/OctopusVersionTests.cs:788]: Ahh I see, the GitHub diff hid that away.

**Review by YuKitsune — APPROVED** [review]
Nice and simple, thanks for implementing this!

**Comment by scme0** [comment]
[sc-64777]

**Pull request merged**